analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/LocksReleasedAllPathsBase.cs

```diff
@@ -49,11 +49,35 @@ public abstract class LocksReleasedAllPathsBase : SymbolicRuleCheck
 
         // ToDo: Implement early bail-out if there's no interesting descendant node in context.Node to avoid useless SE runs
         public override bool ShouldExecute() =>
-            NodeContext.Node.DescendantNodes().OfType<IdentifierNameSyntax>().Any(x => x.Identifier.Text.Contains("Exit") || x.Identifier.Text.Contains("ReleaseReaderLock"));
+            NodeContext.Node.DescendantNodes().OfType<IdentifierNameSyntax>().Any(x => x.Identifier.Text.Contains("Exit")
+                                                                                       || x.Identifier.Text.Contains("ReleaseMutex")
+                                                                                       || x.Identifier.Text.Contains("ReleaseReaderLock"));
 
         public override ProgramState PostProcess(SymbolicContext context)
         {
-            if (context.Operation.Instance.AsInvocation() is { } invocation)
+            if (context.Operation.Instance.AsObjectCreation() is { } objectCreation)
+            {
+                if (objectCreation.Type.Is(KnownType.System_Threading_Mutex)
+                    && context.Operation.Parent.AsAssignment() is { } assignment
+                    && objectCreation.Arguments.Length > 0
```

Comment:
Not as performant, but more readable and aligned with S1155
```suggestion
                    && objectCreation.Arguments.Any()
```

Reply:
I prefer to use `Length` since it's already computed for arrays and we will avoid the extra memory allocation.